File: Binary Search/C++/LC 154 - FIND MINIMUM IN ROTATED SORTED ARRAY II.cpp

// LEET CODE 154 - FIND MINIMUM IN ROTATED SORTED ARRAY II
// QUESTION LINK - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array-ii/


int findMin(vector<int>& nums) {
        int low=0;
        int high=nums.size()-1;

        while(low<high)
        {
            int mid = (low+high)/2;
            if(nums[mid]>nums[high])
            {
                low=mid+1;
            }
	    else if(nums[low]==nums[mid] && nums[mid]==nums[high])
            {
                low++;
                high--;
            }
            else
            {
                high=mid;
            }
            
        }
        return nums[low];
    }

File: Binary Search/Java/LC 154 - FIND MINIMUM IN ROTATED SORTED ARRAY II.java
// LEET CODE 154 - Find Minimum in Rotated Sorted Array II
// QUESTION LINK - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array-ii/

class Solution {
    public int findMin(int[] nums) {
        int low = 0, high = nums.length - 1;

        while (low < high) {
            int mid = (low + high) / 2;

            if (nums[mid] > nums[high]) {
                low = mid + 1;
            } else if (nums[low] == nums[mid] && nums[mid] == nums[high]) {
                low++;
                high--;
            } else {
                high = mid;
            }
        }

        return nums[low];
    }
}
